components/Filters/editor/Preview.js

```diff
@@ -0,0 +1,146 @@
+import React from 'react';
+import { c } from 'ttag';
+import PropTypes from 'prop-types';
+import { Alert, Row, Label, useFormattedLabels } from 'react-components';
+
+import PreviewActionValue from './PreviewActionValue';
+import './Preview.css';
+
+const fomartActions = (actions, labelsModel) => {
+    const getMarkLabel = ({ Starred, Read }) => {
+        if (Starred) {
+            return c('Filter preview').t`starred`;
+        }
+
+        if (Read) {
+            return c('Filter preview').t`read`;
+        }
+    };
+
+    const MAP_FOLDER_ICON = {
+        inbox: 'inbox',
+        archive: 'archive'
+    };
+
+    const getFolderIcon = (folder) => MAP_FOLDER_ICON[folder] || 'folder';
+
+    return Object.keys(actions).reduce((acc, key) => {
+        if (key === 'Mark') {
+            const value = getMarkLabel(actions[key]);
+            value &&
+                acc.push({
+                    label: c('Filter preview').t`Mark email as`,
+                    icon: value === 'starred' ? 'star' : 'read',
+                    value
+                });
+        }
+
+        if (key === 'FileInto') {
+            const map = labelsModel.getLabelsMap();
+            const { folder, labels } = actions[key].concat(actions.Labels || []).reduce(
+                (acc, name) => {
+                    map[name] && acc.labels.push(map[name]);
+                    !map[name] && (acc.folder = name);
+                    return acc;
+                },
+                { folder: '', labels: [] }
+            );
+
+            folder &&
+                acc.push({
+                    label: c('Filter preview').t`Move email to`,
+                    icon: getFolderIcon(folder),
+                    value: folder
+                });
+
+            labels.length &&
+                acc.push({
+                    label: c('Filter preview').t`Label email as`,
+                    icon: 'label',
+                    value: labels
+                });
+        }
+
+        return acc;
+    }, []);
+};
+
+function PreviewFilter({ filter }) {
+    const [labelsModel] = useFormattedLabels();
```

Comment:
```suggestion
    const [labelsModel = []] = useFormattedLabels();
```

Reply:
The hook will always return factory from an Array ;)